Modernization & Bootstrap 5.3.3

TheSeg opened the pull request:
I made some infrastructure updates and some accessibility improvements to the site.

### Bootstrap 5.3.3
* Trimmed down the Bootstrap compile with a custom component import.
* Use Bootstrap CSS variables and functions when available.
* Use Bootstrap's Color Mode functionality.

### Style & Components
* Replaced `.green` and `.light-green` classes with `.current` and `.previous`.
* Made Obsolete & Vintage badges include components.

### Other
* Updated jQuery to 3.7.1
* Updated Clipboard to 2.0.11
* Updated Jekyll to 4.3.4

TheSeg commented:
Hey @ebelinski, checking in if you're still accepting PRs for this project.

Let me know if these changes work out for you, or if you have any questions.

ebelinski commented:
Hi @TheSeg, thank you so much for creating this pull request! I think the modernization changes here are great. I'm sorry I didn't respond earlier. I made some changes this week like upgrading Bootstrap, before I realized that the changes in this PR would conflict with that. Could you please rebase your changes off the latest in `main`? I promise to review/merge it in a timely manner this time.

TheSeg commented:
Merged, updated, and all fixed up! Ready for your review and merging.

* Updated Jekyll to 4.4.1
* Added additional SASS settings: Silence `import` deprecation, declare `_sass` folder, compress CSS output.
* Both iPhone 6 / Plus are now marked as obsolete; removed note.
* Fix missing `<tr>` tags.
* While some pages are not public, changed to new templates & classes.

TheSeg commented:
Checking in on review status. Let me know if you have any questions, especially around the new `.current` and `.previous` classes.

ebelinski commented:
Hey @TheSeg, these changes look good overall. I merged them and working on resolving the deployment failure. Once I do that, I'm going to make some tweaks, I think the link color and page width changed with these changes. I'll let you know when I'm done!

TheSeg commented:
Apologies if you already know this, but you can modify these settings at `css/_variables.scss`.

ebelinski commented:
Thanks for the heads up, I made some modifications and imported the variables file before bootstrap in order to have the custom variables override the variables in bootstrap.
